File: src/gestorAplicación/clasesHeredadas/Trabajador.java
package gestorAplicación.clasesHeredadas;


import java.util.ArrayList;
import gestorAplicación.clasesPrincipales.*;

// Implementación de características de POO
// Herencia

public class Trabajador extends Persona {
    final private String horario;
    private ArrayList<Reserva> mesasAtendidas = new ArrayList<Reserva>();
    public static ArrayList<Trabajador> trabajadoresActivos = new ArrayList<Trabajador>();
    public static ArrayList<Reserva> mesasElegir = new ArrayList<Reserva>();
    private int sueldo;

// Implementación de características de POO
// Sobrecarga de constructores

    public Trabajador(String nombre, int id, String horario){
        super(nombre, id);
        this.horario = horario;
    }

    public Trabajador(int id, String horario) {
        super("N/N", id);
        this.horario = horario;
    }
    public static ArrayList<Trabajador> getTrabajadoresActivos() {
        return trabajadoresActivos;
        }  
    public static void addTrabajadoresActivos(Trabajador trabajador) {
        trabajadoresActivos.add(trabajador);
        }
    public static void removeTrabajadoresActivos(Trabajador trabajador) {
        trabajadoresActivos.remove(trabajador);
        }

    public static ArrayList<Reserva> getMesasElegir() {
        return mesasElegir;
        }   
    public static void addMesasElegir(Reserva reserva) {
        mesasElegir.add(reserva);
    }
    public static void removeMesasElegir(Reserva reserva) {
        mesasElegir.remove(reserva);
        }
    public String toString() {
        return this.getNombre() + this.getId() + this.horario;
    }
    
    public int getSueldo() {
        return sueldo;
        }
    public void addMesasAtendidas(Reserva reserva) {
            mesasAtendidas.add(reserva);
    }
    public ArrayList<Reserva> getMesasAtendidas() {
        return mesasAtendidas;
        }  


    @Override
     public void accion(Factura facturaNow, int sueldo){
        Reserva reserva = facturaNow.getReserva();
        addMesasAtendidas(reserva);
        System.out.println("La cantidad de reservas atendidas por el trabajador: " + getNombre());
        System.out.println("Es un total de: " + mesasAtendidas.size());
        

        int sueldoFactura = facturaNow.getPrecio();
        this.sueldo += sueldoFactura/2;
        System.out.println("Su sueldo actual es de: " + getSueldo());
     }

}

File: Python/src/gestorAplicación/clasesHeredadas/Trabajador.py
from gestorAplicación.clasesPrincipales import Persona

class Trabajador(Persona):
    _mesasAtendidas = []
    trabajadoresActivos = []
    mesasElegir = []

    def __init__(self, id, horario, nombre = "N/N"):
        super().__init__(nombre, id)
        self._horario = horario
    
    #Métodos get
    @classmethod
    def getTrabajadoresActivos():
        return Trabajador.trabajadoresActivos
    
    @classmethod
    def getMesasElegir():
        return Trabajador.mesasElegir
    
    @classmethod
    def getMesasAtendidas():
        return Trabajador._mesasAtendidas
    
    def getSueldo(self):
        return self._sueldo
    
    #Otros métodos
    @classmethod
    def addTrabajadoresActivos(trabajador):
        Trabajador.trabajadoresActivos.append(trabajador)
    
    @classmethod
    def removeTrabajadoresActivos(trabajador):
        Trabajador.trabajadoresActivos.remove(trabajador)
    
    @classmethod
    def addMesasElegir(reserva):
        Trabajador.mesasElegir.append(reserva)
    
    @classmethod
    def removeMesasElegir(reserva):
        Trabajador.mesasElegir.remove(reserva)

    @classmethod
    def addMesasAtendidas(reserva):
        Trabajador.addMesasAtendidas.append(reserva)
    
    #Método ToString
    def __str__(self):
        return self.getNombre() + self.getId() + self._horario

    #Sobreescritura de método (Override) 
    def accion(self, facturaNow, sueldo):
        reserva = facturaNow.getReserva()
        Trabajador.addMesasAtendidas(reserva)

        sueldoFactura = facturaNow.getPrecio()
        self._sueldo += sueldoFactura/2
        return True